lib/RedisRemoteSaiInterface.cpp

```diff
@@ -1375,6 +1365,113 @@ sai_status_t RedisRemoteSaiInterface::waitForGetStatsResponse(
     return status;
 }
 
+sai_status_t RedisRemoteSaiInterface::queryStatsCapability(
+        _In_ sai_object_id_t switchId,
+        _In_ sai_object_type_t objectType,
+        _Inout_ sai_stat_capability_list_t *stats_capability)
+{
+    SWSS_LOG_ENTER();
+
+    auto switchIdStr = sai_serialize_object_id(switchId);
+    auto objectTypeStr = sai_serialize_object_type(objectType);
+
+    if (stats_capability == NULL)
+    {
+        SWSS_LOG_ERROR("Failed to find stats-capability: switch %s, object type %s", switchIdStr.c_str(), objectTypeStr.c_str());
+        return SAI_STATUS_INVALID_PARAMETER;
+    }
+
+    if (stats_capability && stats_capability->list)
```

Comment:
need to check if list is null and count non zero, this is invalid scenario

Reply:
also if stats_capability is NULL, then you skip if and in line 1387 you will have exception for page fault

Reply:
I have added count non-zero check.  if stats_capability is NULL, this function will return from line 1381 only.